python/sample.py

```diff
@@ -0,0 +1,96 @@
+import json
+import urllib.request
+import urllib.parse
+import sys
+import os
+import argparse
+import winreg as wreg
+
+def get_scheduled_events(address, headers):
+    '''
+    Make a GET request and return the JSON received.
+    '''
+    request = urllib.request.Request(address, headers=headers)
+    response = urllib.request.urlopen(request)
+    return json.loads(response.read())
+
+def post_scheduled_events(address, data, headers):
+    '''
+    Make a POST request.
+    '''
+    request = urllib.request.Request(address, data=str.encode(data), headers=headers)
+    urllib.request.urlopen(request)
+
+def get_ip_address(args):
+    '''
+    Get the IP address of scheduled event. Returns None if IP address is not provided or stored.
+    '''
+    ip_address = None
+    if args.ip_address:
+        # use IP address provided in parameter
+        ip_address = args.ip_address
+    elif args.use_registry and sys.platform == 'win32':
+        # use CloudControlIp in registry if available.
+        try:
+            key = wreg.OpenKey(wreg.HKEY_LOCAL_MACHINE, "Software\\CloudControl")
+            ip_address = wreg.QueryValueEx(key, 'CloudControlIp')[0]
+            key.Close()
+        except FileNotFoundError:
+            return None
+    elif sys.platform == 'win32' or "linux" in sys.platform:
+        # use CLOUDCONTROLIP in system variables if available.
+        ip_address = os.getenv('COUDCONTROLIP')
+    return ip_address
+
+def parse_args():
+    '''
+    Parse command line arguments.
+    '''
+    parser = argparse.ArgumentParser(description="Sample code for getting scheduled events.")
+    parser.add_argument('--use_registry', action="store_true",
+                        help="Get the IP address from Windows registry")
+    parser.add_argument('--ip_address', type=str,
+                        help="The IP address of scheduled events endpoint.")
```

Comment:
Good idea! Although I think this is already incorporated into the code in the function get_ip_address. It first tries to look into registry (if the parameter is provided) and then look into the environment variable.